HW3 is completed

Elyas-Crimean opened the pull request:
## Домашнее задание №3 «Частотный анализ»

### Чек-лист студента ([Что это?](https://github.com/OtusGolang/home_work/wiki/%D0%9A%D0%BE%D0%BC%D0%BC%D0%B5%D0%BD%D1%82%D0%B0%D1%80%D0%B8%D0%B8-%D0%BA-%D1%87%D0%B5%D0%BA-%D0%BB%D0%B8%D1%81%D1%82%D1%83-%D1%81%D1%82%D1%83%D0%B4%D0%B5%D0%BD%D1%82%D0%B0))
- [ ] Я перечитал задание после решения. [Зачем перечитывать задание?](https://github.com/OtusGolang/home_work/wiki/%D0%9A%D0%BE%D0%BC%D0%BC%D0%B5%D0%BD%D1%82%D0%B0%D1%80%D0%B8%D0%B8-%D0%BA-%D1%87%D0%B5%D0%BA-%D0%BB%D0%B8%D1%81%D1%82%D1%83-%D1%81%D1%82%D1%83%D0%B4%D0%B5%D0%BD%D1%82%D0%B0#user-content-%D0%97%D0%B0%D1%87%D0%B5%D0%BC-%D0%BF%D0%B5%D1%80%D0%B5%D1%87%D0%B8%D1%82%D1%8B%D0%B2%D0%B0%D1%82%D1%8C-%D0%B7%D0%B0%D0%B4%D0%B0%D0%BD%D0%B8%D0%B5)
- [ ] Я запустил `go mod tidy`.
- [ ] Я удалил `.sync` файл. [Зачем его удалять?](https://github.com/OtusGolang/home_work/wiki/%5B%D0%A1%D1%82%D1%83%D0%B4%D0%B5%D0%BD%D1%82%D0%B0%D0%BC%5D-%D0%9F%D1%80%D0%BE%D1%86%D0%B5%D1%81%D1%81-%D1%81%D0%B4%D0%B0%D1%87%D0%B8-%D0%94%D0%97#user-content-%D0%92%D0%B0%D1%80%D0%B8%D0%B0%D0%BD%D1%82-2)

### Критерии оценки
- [ ] Пайплайн зелёный - 4 балла
- [ ] Добавлены новые юнит-тесты - до 4 баллов
- [ ] Понятность и чистота кода - до 2 баллов

#### Зачёт от 7 баллов

Elyas-Crimean committed:
HW3 is completed

engineer-go commented on hw03_frequency_analysis/top.go:
```diff
@@ -1,6 +1,51 @@
+
+var (
+	startPunctuations = regexp.MustCompile(`^(\.|,|<|\/|:|!|>|-|;)+`)
+	endPunctuations   = regexp.MustCompile(`(\.|,|<|\/|:|!|>|-|;)+$`)
```
Вот это не совсем корректно. Т.к. у меня в тестах есть к примеру символ » и это тут же ломает вашу логику. Ну т.е. либо тут вам весь utf загонять с его символами, либо всё же крафтить более умную логику, которая ожидает всё что угодно кроме буквы собственно.

Elyas-Crimean committed:
HW3 is completed step2

так наверно будет красивее, хотя в идеале следует уточнять
 условия в зависимости от текста и языка

engineer-go commented on hw03_frequency_analysis/top.go:
```diff
@@ -1,6 +1,51 @@
+	"strings"
+)
+
+type Counter struct {
+	word  string
+	count int
+}
```
А не подскажете, почему вы не использовали ту же мапу? Вроде бы как функционал не отличается?

Elyas-Crimean committed:
HW3 is completed step3
